crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -423,7 +434,7 @@ struct ActiveFilter {
 enum FilterKind {
     Log(Box<Filter>),
     Block,
-    PendingTransaction,
+    PendingTransaction(Arc<Mutex<Receiver<TxHash>>>),
```

Comment:
I want to move this to a new type that wraps the Arc Mutex and has an `async fn drain()` that returns the new hashes

Reply:
Ok! I introduced a new type